xcell/tests/test_data.py

```diff
@@ -0,0 +1,237 @@
+        if tr.split('__')[0] in trs:
+            tracers_for_cl_test.append(tr)
+
+    assert tracers_for_cl == tracers_for_cl_test
+    assert data.get_tracers_used(True) == ['DESgc__0', 'DESwl__0', 'DESwl__1',
+                                           'DESwl__2', 'DESwl__3', 'PLAcv']
+    remove_yml_file(data.data)
+
+
+def test_get_tracer_bare_name():
+    data = get_data()
+    config = data.data
+
+    for tr in config['tracers']:
+        bname = data.get_tracer_bare_name(tr)
+        assert tr.split('__')[0] == bname
+    remove_yml_file(data.data)
+
+
+def test_get_cl_trs_names():
+    data = get_data()
+    config = data.data
+
+    def assert_cls(wsp=False):
+        cl_trs = data.get_cl_trs_names(wsp)
+
+        cl_trs_test = []
+        trs = data.get_tracers_used(wsp)
+        for i, tri in enumerate(trs):
+            for trj in trs[i:]:
+                bn1 = data.get_tracer_bare_name(tri)
+                bn2 = data.get_tracer_bare_name(trj)
+                key = '-'.join([bn1, bn2])
+                if (config['cls'][key]['compute'] == 'auto') and (tri == trj):
+                    cl_trs_test.append((tri, trj))
+                elif config['cls'][key]['compute'] == 'all':
+                    cl_trs_test.append((tri, trj))
+
+        assert cl_trs == cl_trs_test
+
+    assert_cls()
+    assert_cls(True)
+    remove_yml_file(config)
+
+
+def test_get_cov_trs_name():
+    data = get_data()
+
+    def assert_cls(wsp=False):
+        cl_trs = data.get_cl_trs_names(wsp)
+
+        cov_trs_test = []
+        for i, trsi in enumerate(cl_trs):
+            for trsj in cl_trs[i:]:
+                cov_trs_test.append((*trsi, *trsj))
+
+        assert cov_trs_test == data.get_cov_trs_names(wsp)
+
+    assert_cls()
+    assert_cls(True)
+    remove_yml_file(data.data)
+
+
+def test_get_cov_ng_cl_tracers():
+    data = get_data()
+    config = data.data
+    cl_trs = data.get_cl_trs_names()
+
+    d = {}
+    for trs in config['cov']['ng']['order']:
+        d[trs] = []
+
+    for trs in cl_trs:
+        bn1 = data.get_tracer_bare_name(trs[0])
+        bn2 = data.get_tracer_bare_name(trs[1])
+        key = '-'.join([bn1, bn2])
+        if key in d:
+            d[key].append(trs)
+        else:
+            key = '-'.join([bn2, bn1])
+            d[key].append(trs)
+
+    # Order the tracers so that gc0 - wl0, gc1 - wl0, etc.
+    d['DESwl-DESgc'].sort(key=lambda x: x[1])
+
+    cl_trs_cov = []
+    for trs in config['cov']['ng']['order']:
+        cl_trs_cov.extend(d[trs])
+
+    assert cl_trs_cov == data.get_cov_ng_cl_tracers()
+    remove_yml_file(config)
+
+
+def test_filter_tracers_wsp():
+    data = get_data()
+    config = get_config_dict()
+
+    all_tracers = list(config['tracers'].keys())
+    trs_wsp = []
+    trs_mask = []
+    for tr, val in config['tracers'].items():
+        mask_name = val['mask_name']
+        if mask_name not in trs_mask:
+            trs_wsp.append(tr)
+            trs_mask.append(mask_name)
```

Comment:
are you using `trs_mask` for anything?

Reply:
Yes, to keep track of the mask names already taken into account by a previous tracer.

Reply:
oh, right, sorry.